Description: adapt test-ci build target for buildd
 * log to stdout
 * increase timeout multipliers
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not-needed
Reviewed-By: Xavier Guimard <yadd@debian.org>
Last-Update: 2020-02-09

--- a/Makefile
+++ b/Makefile
@@ -558,7 +558,7 @@
 # This target should not use a native compiler at all
 # Related CI job: node-test-commit-arm-fanned
 test-ci-js: | clear-stalled
-	$(PYTHON) tools/test.py $(PARALLEL_ARGS) -p tap --logfile test.tap \
+	$(PYTHON) tools/test.py $(PARALLEL_ARGS) -p tap \
 		--mode=$(BUILDTYPE_LOWER) --flaky-tests=$(FLAKY_TESTS) \
 		--skip-tests=$(CI_SKIP_TESTS) \
 		$(TEST_CI_ARGS) $(CI_JS_SUITES)
--- a/test/common/index.js
+++ b/test/common/index.js
@@ -278,16 +278,15 @@
 
 function platformTimeout(ms) {
   const multipliers = typeof ms === 'bigint' ?
-    { two: 2n, four: 4n, seven: 7n } : { two: 2, four: 4, seven: 7 };
+    { fast: 2n, slow: 4n } : { fast: 3, slow: 5 };
+ 
+  if (process.arch.startsWith('arm') || process.arch.startsWith('mips') || process.arch.startsWith('riscv'))
+    ms = multipliers.slow * ms;
+  else
+    ms = multipliers.fast * ms;
 
   if (process.features.debug)
-    ms = multipliers.two * ms;
-
-  if (exports.isAIX || exports.isIBMi)
-    return multipliers.two * ms; // Default localhost speed is slower on AIX
-
-  if (isPi)
-    return multipliers.two * ms;  // Raspberry Pi devices
+    ms = multipliers.slow * ms;
 
   return ms;
 }
--- a/tools/test.py
+++ b/tools/test.py
@@ -1762,7 +1762,8 @@
 
   def should_keep(case):
     if any((s in case.file) for s in options.skip_tests):
-      return False
+      case.outcomes.add(FLAKY)
+      return True
     elif SKIP in case.outcomes:
       return False
     elif (options.flaky_tests == SKIP) and (set([SLOW, FLAKY]) & case.outcomes):
--- a/test/sequential/sequential.status
+++ b/test/sequential/sequential.status
@@ -8,7 +8,7 @@
 # https://github.com/nodejs/node/issues/27611#issuecomment-613100468
 test-cpu-prof-dir-worker: PASS, FLAKY
 # https://github.com/nodejs/node/issues/44898
-test-watch-mode: PASS, FLAKY
+test-watch-mode: SKIP
 test-watch-mode-inspect: PASS, FLAKY
 # https://github.com/nodejs/node/issues/47409
 test-http2-large-file: PASS, FLAKY
